org.adempiere.base/src/org/adempiere/util/ModelInterfaceGenerator.java

```diff
@@ -792,9 +792,7 @@ public static void generateSource(String sourceFolder, String packageName, Strin
 		if (tableName == null || tableName.trim().length() == 0)
 			throw new IllegalArgumentException("Must specify table name");
 
-		StringBuilder tableLike = new StringBuilder().append(tableName.trim());
-		if (!tableLike.toString().startsWith("'") || !tableLike.toString().endsWith("'"))
-			tableLike = new StringBuilder("'").append(tableLike).append("'");
+		StringBuilder tableLike = new StringBuilder().append(tableName.trim().toUpperCase());

```

Comment:
@nmicoud 

Can you please check?

There are differences between the methods generateSource of ModelClassGenerator vs ModelInterfaceGenerator

In theory these two methods must be the same with exception of the line within the while ... new Model

Ideally, because this is duplicated code, it would be better to have it in just one place.

For example, if we can call ModelClassGenerator.generateSource from ModelInterfaceGenerator.

This line here is very relevant, if you don't remove the quotes, then it will fail on line 848 when receiving a quoted tablename

As we are dealing with tablenames, I think is better also to remove spaces too, with the previous approach spaces in the tableLike were allowed, but with this change, spaces will be problematic.

Reply:
Hi @CarlosRuiz-globalqss ,

Something like that?
Not sure about the name of the new generateSource method with type parameter (which is not yet in javadoc).

And I don't understand what you suggest with spaces ; can you give an example where it can fail?

Thanks,

Reply:
> And I don't understand what you suggest with spaces ; can you give an example where it can fail?

I haven't tested, but from the code I think before the patch the following will work:
`'AD_Table', 'AD_Column'`   <-  (see the space after the comma)
but after the patch I think it will fail because you are removing the quotes and adding them back, but not removing the spaces

Reply:
Just made a test and both tables were considered